application/libraries/Api/Command/V1/SurveyPatch/OpHandlerQuestionTrait.php

```diff
@@ -288,4 +293,45 @@ private function getScaleIdFromData(array $questionData)
             $questionData
         ) ? (int)$questionData['scale_id'] : 0;
     }
+
+    /**
+     * Maps the tempIds of new subquestions or answers to the real ids.
+     * @param Question $question
+     * @param array $data
+     * @param bool $answers
+     * @return array
+     */
+    private function getSubQuestionNewIdMapping(
+        Question $question,
+        array $data,
+        bool $answers = false
+    ): array {
+        $tempIds = [];
+        $mapping = [];
+        $title = $answers ? 'code' : 'title';
+        $object = $answers ? 'answers' : 'subquestions';
+        $idField = $answers ? 'aid' : 'qid';
+        foreach ($data as $subQueDataArray) {
+            foreach ($subQueDataArray as $subQueData) {
+                if (
+                    isset($subQueData['tempId'])
+                    && isset($subQueData[$title])
+                ) {
+                    $tempIds[$subQueData[$title]] = $subQueData['tempId'];
+                }
+            }
+        }
+        if (count($tempIds) > 0) {
+            $question->refresh();
+            foreach ($question->$object as $subquestion) {
+                if (array_key_exists($subquestion->$title, $tempIds)) {
+                    $mapping[$object . 'Map'][] = [
```

Comment:
This value in this array could be an object. Maybe create a class with public properties. An objects public properties are represented in the JSON output:

```new TempIdMapItem('someTempId123', 'newRealId', 'fieldName')```

The benefit of this is that the next developer that needs to add a temp id mapping doesn't need to know the data structure to return. They just create one of these objects. Also its less likely that a make will be made in the data structure.

``` 
class TempIdMapItem
{
    public $tempId = null;
    public $id = null;
    public $field = 'id';
    
    public function __construct($tempId , $id, $field = 'id')
    {
        $this->tempId = $tempId;
        $this->id = $id;
        $this->field = $field;
    }
}
```